tests/test_datastores.py

```diff
@@ -258,6 +258,25 @@ def test_forbidden_file_access():
         obj.stat()
 
 
+def test_verify_data_stores_are_not_cached_in_api():
```

Comment:
```suggestion
def test_verify_data_stores_are_not_cached_in_api_when_not_needed():
```

I think we also need a test to verify what happens when not passing secrets and running on server side - mock the `_executing_on_client` function (another reason why it was good to separate it to a function 😃) and don't pass store-secrets, verify caching still doesn't happen.